src/main/scala/chisel3/experimental/util/algorithm/Bitwise.scala

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental.util.algorithm
+
+import chisel3._
+
+/** Map each bits to logical or of itself and all bits less siginificant than it.
+  * @example {{{
+  * LSBOr("b00001000".U) // Returns "b11111000".U
+  * LSBOr("b00010100".U) // Returns "b11111100".U
+  * LSBOr("b00000000".U) // Returns "b00000000".U
+  * }}}
+  * This circuit seems to be high fan out, but synthesis tool should handle this.
+  */
+object LSBOr {
```

Comment:
It should be renamed to `leftOr` to coordinate to RC.

Reply:
IMO the naming leftOr has some discrepancy with "left" in "shift left". `leftOr` is actually or-ing all bits in the right (LSB). This can be a source of confusion. Actually I got it wrong when adding cross-check with rocket-core's implementation.

BTW, if we really want to keep in line with RC's naming, should we use `leftOR` instead of `leftOr`?

Reply:
> BTW, if we really want to keep in line with RC's naming, should we use leftOR instead of leftOr?

yeah, Chisel use `Or` everywhere, so `OR` is an anti-pattern in Chisel(but not in RC).

> `leftOr` is actually or-ing all bits in the right (LSB). This can be a source of confusion.

I agree, that is confusing as well(actually every time I use this function I need to take a look at its implementation). Personally I like the `lsbOr` better, which means `OR` LSB data of each bits. I think for the naming part, we may not need to align with RC.
what's opinion from @jackkoenig and @ekiwi? I'm OK at `lsbOr` and `msbOr` for this function.

Reply:
My suggestion here is: make a PR to rocket-chip to change the name of their functions from `leftOR` to `lsbOr` and from `rightOR` to `msbOr`. As soon as that PR is approved, we can then go ahead and add the function to chisel3. If the rocket-chip folks do not approve of changing the name, that means that suddenly there are two copies of the same function, one in rocket-chip and one in chisel.

Reply:
h/t @seldridge for some archeology, the original name of this was `highOR`, then changed to `leftOR` when `rightOR` was added. Food for thought.

Reply:
From dev today, our current favorite seems to be `prefixOrFromLsb` (and `prefixOrFromMsb` for the other one).

Reply:
I chatted about this a bit with @terpstra. He argued that `fromLsb` is clear, although arguably `LsbFirst` is another alternative to consider. That being said, he argued (and I agree) that `left` and `right` are actually unambiguous. It is no different than a `foldLeft`, `reduceLeft`, whatever. It is a bit unfortunately that number like `54321` decomposes to `List(1, 2, 3, 4, 5)`, but this is universal. Iterate on the bits of a bitvector in Verilog and you get the exact same thing.

All of this is to say, I now think that `prefixOrLeft` and `prefixOrRight` are perfectly acceptable options. I think they are a lot better than `leftOr` because, (1) it matches the style of `foldLeft`, and (2) `prefixOr` is much more clearly the operation than just an `Or`. With that in mind, I'll accept `prefixOrFromLsb`, but I actually prefer `prefixOrLeft` now.

Reply:
:+1:  on `prefixOrLeft`

Reply:
Note: `prefixOr` is a made-up name that doesn't appear to have prior art.  The line of reasoning here appears to be this looks like a `prefixSum` and since sum is `xor` in GF(2), then it's reasonable to call this `prefixXor`.  If we use a different reduction, then it's reasonable to call this `prefix<binary-operator>`.  I would prefer to not invent our own thing here and stick with "scan" which Wikipedia (without any citations) calls the generalization of `prefixSum` to other binary operators.

Reply:
😅 And concretely, I'm proposing `scanLeftOr`.

Reply:
My argument in favor of using `Left` relies on the universal naming convention. This makes my arguing for `prefixOr` is a bit inconsistent since we made that one up. `scanLeftOr` is probably the most "conventionally consistent name), since a reasonable implementation of this behavior uses it:
ie. 
```scala
data.asBools.scanLeft(false.B)(_ || _).tail.asUInt
```
This is really just a special scanLeft on the bits of a `UInt`...

Reply:
I think calling it a `scan` is appropriate.  (It's less immediately obvious what `prefixOr` means, and it kind of sounds like it's going to increase the width of the input by prefixing something to it.)

Although I agree there's a case to be made for `left` and `right`, it's inevitably going to confuse people, because we are defining this on UInt, and the leftmost part of a UInt, when written out, is the MSB.

Reply:
@aswaterman had an interesting suggestion, what about `Up` and `Down`? While I think the convention of `Left` and `Right` is unambiguous on sequences, and I am comfortable extending the convention to bit vectors like `UInt`, I think this is an interesting idea.

So, yet another proposal is the convention of "Up" to mean from least-significant bit from most-significant bit, and "Down" to mean the opposite. Thus the name of this function could be: `scanUpOr`.

My main concern about "Up" and "Down" is that it would be inconsistent with the methods on `Vec` which I really do not think should deviate from the normal functional convention given that Vecs are Scala Seqs. There would be a little bit of extra convention learning that `scanUp` is equivalent to `.asBools.scanLeft`, but I think that's reasonable if there's general consensus that "Up"/"Down" is much more clear than "Left"/"Right" in the context of bit vectors like `UInt`.

Reply:
The renaming is done in e6fab506430c1bce4ce5e0de64cba8e08658d3f6